Build errors due to ptrdiff_t and missing/misplaced stddef.h

## Background information
I encounter `unknown type name ‘ptrdiff_t’` errors due to missing/misplaced `#include <stddef.h>` in the developer build from a git checkout

### What version of Open MPI are you using?
master / 76eaffa6b8017999b0d014714ee29a37ca1d60b8

git submodule status 
+517d954a4a3929fe27e69f35cbe00a419c143dbf 3rd-party/openpmix (v1.1.3-2836-g517d954a)
+67addf25843e7765d6aee19b51a3165d08a9eace 3rd-party/prrte (dev-30952-g67addf2584)

### Please describe the system on which you are running
* Operating system/version: Arch Linux
* Computer hardware: AMD/x86_64
* autotolls:  autoconf 2.71,  libtool 2.4.6.42,  automake 1.16.3
* gcc 10.2.0 and clang 11.1.0

## Details of the problem

>   CC       libompi_dbg_msgq_la-ompi_msgq_dll.lo
> In file included from ompi/ompi/ompi/include/ompi_config.h:29,
>                  from ompi/ompi/ompi/debuggers/ompi_msgq_dll.c:72:
> ../../opal/include/opal_config.h:1317:28: error: unknown type name ‘ptrdiff_t’
>  1317 | #define OMPI_MPI_AINT_TYPE ptrdiff_t
>       |                            ^~~~~~~~~
> ../../ompi/include/mpi.h:402:9: note: in expansion of macro ‘OMPI_MPI_AINT_TYPE’
>   402 | typedef OMPI_MPI_AINT_TYPE MPI_Aint;
>       |         ^~~~~~~~~~~~~~~~~~

The `#include <stddef.h>` required for `ptrdiff_t` is missing here.

Inserting `stddef.h` by hand to the top of `ompi_config.h` triggers the related error in `mpi.h`:

>   CC       mpi-io/close.lo
> In file included from ompi/ompi/ompi/mca/io/romio321/romio/adio/include/adio.h:72,
>                  from ompi/ompi/ompi/mca/io/romio321/romio/mpi-io/mpioimpl.h:17,
>                  from ompi/ompi/ompi/mca/io/romio321/romio/mpi-io/close.c:8:
> ./../../../../../ompi/include/mpi.h:157:28: error: unknown type name ‘ptrdiff_t’
>   157 | #define OMPI_MPI_AINT_TYPE ptrdiff_t
>       |                            ^~~~~~~~~
> ./../../../../../ompi/include/mpi.h:402:9: note: in expansion of macro ‘OMPI_MPI_AINT_TYPE’
>   402 | typedef OMPI_MPI_AINT_TYPE MPI_Aint;
>       |         ^~~~~~~~~~~~~~~~~~
> In file included from ompi/ompi/ompi/mca/io/romio321/romio/adio/include/adio.h:72,
>                  from ompi/ompi/ompi/mca/io/romio321/romio/mpi-io/mpioimpl.h:17,
>                  from ompi/ompi/ompi/mca/io/romio321/romio/mpi-io/close.c:8:
> ./../../../../../ompi/include/mpi.h:436:5: error: unknown type name ‘size_t’
>   436 |     size_t _ucount;
>       |     ^~~~~~

In `mpi.h.in` the required `stddef.h` is included too late and guarded by some OPAL_STDC_HEADERS:
> /* type to use for ptrdiff_t, if it does not exist, set to ptrdiff_t if it does exist */
> #undef ptrdiff_t
> ..
> /* include for ptrdiff_t */
> #ifdef OPAL_STDC_HEADERS
> #include <stddef.h>
> #endif

Not clear how to fix this to me as `OMPI_MPI_AINT_TYPE` is set via `AC_DEFINE_UNQUOTED` and there are several *h.in files as well as a `AH_TOP` header in `configure.ac`

--- a/configure.ac
+++ b/configure.ac
@@ -1366,10 +1366,13 @@ OPAL_SETUP_WRAPPER_FINAL
 # autoconf macro defines in mpi.h.  Since AC sometimes changes whether
 # things are defined as null tokens or an integer result, two projects
 # with different versions of AC can cause problems.
-if test $ac_cv_header_stdc = yes; then
-    AC_DEFINE(OPAL_STDC_HEADERS, 1,
-              [Do not use outside of mpi.h.  Define to 1 if you have the ANSI C header files.])
-fi
+
+# According to the autoconf 2.67 documentation the AC_HEADER_STDC macro,
+# and therefore the ac_cv_header_stdc cache variable, is obsolescent, as
+# current systems have conforming header files. Instead of removing the
+# protection completely, let's just make sure it is always on.
+AC_DEFINE(OPAL_STDC_HEADERS, 1,
+          [Do not use outside of mpi.h.  Define to 1 if you have the ANSI C header files.])
 if test $ac_cv_header_sys_time_h = yes ; then
     AC_DEFINE(OPAL_HAVE_SYS_TIME_H, 1,
               [Do not use outside of mpi.h.  Define to 1 if you have the <sys/time.h> header file.])

